library/std/src/sys/anonymous_pipe/mod.rs

```diff
@@ -0,0 +1,146 @@
+//! Module for anonymous pipe
```

Comment:
The public docs and types should be in a top level module `src/pipe.rs` rather than in sys.

Reply:
Yep I could do that, but I have to leave the FromRawFd impl in the sys, because only sys allows use of `cfg(unix)`

Reply:
I would then have to leave the unix/windowe/unsupported and the test module in sys, while moving pipe into src

Is there any better filesystem layout?

Reply:
`sys` should expose a single interface across all platforms so in this case it just needs `AnonPipe` and `pipe`. This makes it obvious what platforms have to provide and means that people working in the non-sys parts of std don't have to wade into platform specific details. Keeping things in `sys/anonymous_pipe.rs` is fine.